asyncapi.yml

```diff
@@ -0,0 +1,237 @@
+          - $ref: "#/components/messages/delete_peer"
+          - $ref: "#/components/messages/init"
+
+components:
+  messages:
+    add_rule:
+      summary: Adds an egress rule to the Gateway.
+      payload:
+        $ref: "#/components/schemas/add_rule"
+      examples:
+        - payload:
+            add_rule:
+              dst: 0.0.0.0/8
+              port_range:
+                range_start: 80
+                range_end: 81
+                protocol: tcp
+              user_uuid: 3118158c-29cb-47d6-adbf-5edd15f1af17
+    delete_rule:
+      summary: Removes an egress rule from the Gateway.
+      payload:
+        $ref: "#/components/schemas/delete_rule"
+      examples:
+        - payload:
+            delete_rule:
+              dst: 0.0.0.0/8
+              port_range:
+                range_start: 80
+                range_end: 81
+                protocol: tcp
+              user_uuid: 3118158c-29cb-47d6-adbf-5edd15f1af17
+    add_peer:
+      summary: Adds a peer to the receiver's WireGuard configuration.
+      payload:
+        $ref: "#/components/schemas/add_peer"
+      examples:
+        - name: Gateway example
+          payload:
+            init:
+              default_action: deny
+              interface:
+                address:
+                  - 100.64.11.22/10
+                mtu: 1280
+              peers:
+                - allowed_ips:
+                    - 100.64.11.22/32
+                  public_key: AxVaJsPC1FSrOM5RpEXg4umTKMxkHkgMy1fl7t1xNyw=
+                  preshared_key: LZBIpoLNCkIe56cPM+5pY/hP2pu7SGARvQZEThmuPYM=
+                  user_uuid: 3118158c-29cb-47d6-adbf-5edd15f1af17
+    delete_peer:
+      summary: Removes a peer from the receiver's WireGuard configuration.
+      payload:
+        $ref: "#/components/schemas/delete_peer"
+      examples:
+        - payload:
+            delete_peer:
+              public_key: AxVaJsPC1FSrOM5RpEXg4umTKMxkHkgMy1fl7t1xNyw=
+
+    init:
+      summary: A control command to initialize the gateway with initial config
+      description: |
+        This command initializes a gateway with its configuration after successfully connecting the WebSocket.
+      payload:
+        $ref: "#/components/schemas/init"
+      examples:
+        - name: Gateway example
+          payload:
+            init:
+              default_action: deny
+              interface:
+                address:
+                  - 100.64.11.22/10
+                mtu: 1280
+              peers:
+                - allowed_ips:
+                    - 100.64.11.22/32
+                  public_key: AxVaJsPC1FSrOM5RpEXg4umTKMxkHkgMy1fl7t1xNyw=
+                  preshared_key: LZBIpoLNCkIe56cPM+5pY/hP2pu7SGARvQZEThmuPYM=
+                  user_uuid: 3118158c-29cb-47d6-adbf-5edd15f1af17
+
+  schemas:
+    init:
+      summary: Gateway initialization
+      description: |
+        This message indiciates the gateway should reconfigure itself with the contained configuration.
```

Comment:
Will this be the init only for the gateway and not for the client?

Reply:
Yes. Client api will be synchronous and documented with openapi